### ISSUE TITLE
Marmiton.org is not supported

### ISSUE THREAD

**Issue by sephrat** [mrzapp/nextcloud-cookbook#99]
Marmiton.org is a very popular recipe website in the French community (top result in Google search). Importing recipes from this site does not work.

I'm not specialized in web development so I don't really know what's wrong with this website's schema.org implementation. However something definitely points out it's trying to implement it, but it seems encapsulated in some JS code. I guess it's related to the quantities being calculated dynamically according to the servings you want.

Example: https://www.marmiton.org/recettes/recette_cookies-de-pauline_64092.aspx
```
<script type="application&#x2F;ld&#x2B;json">
--
  |  
  | {"@context":"http://schema.org","@type":"Recipe","name":"Cookies de Pauline","recipeCategory":"cookie","image":"https://assets.afcdn.com/recipe/20131009/24194_w1024h768c1cx3012cy2279.jpg","datePublished":"2007-11-30T04:07:00+01:00","prepTime":"PT20M","cookTime":"PT10M","totalTime":"PT30M","recipeYield":"4 personnes","recipeIngredient":["100 g de sucre roux","150 g de chocolat \u00e0 cuire","1  oeuf","225 g de farine","75 g de sucre blanc","1 paquet de levure","100 g de beurre","1 pinc\u00e9e de sel"],"recipeInstructions":[{"@type":"HowToStep","text":"Tapissez la plaque du four de papier aluminium et la beurrer (ou mettre une feuille de papier sulfuris\u00e9."},{"@type":"HowToStep","text":"M\u00e9langer l'oeuf, le sucre roux et le sucre blanc. Ajouter la pinc\u00e9e de sel."},{"@type":"HowToStep","text":"Faire fondre le beurre. L'ajouter, ainsi que la levure et la farine."},{"@type":"HowToStep","text":"M\u00e9langer tous les ingr\u00e9dients avec une cuill\u00e8re en bois."},{"@type":"HowToStep","text":"Casser le chocolat en petits morceaux et l'ajouter."},{"@type":"HowToStep","text":"M\u00e9langer le tout avec les mains."},{"@type":"HowToStep","text":"Mettre au r\u00e9frig\u00e9rateur pendant 30 minutes."},{"@type":"HowToStep","text":"Pr\u00e9chauffer le four \u00e0 240\u00b0C (thermostat 8)."},{"@type":"HowToStep","text":"Pr\u00e9parer les boulettes (+ les boulettes sont grosses plus les cookies seront grands)."},{"@type":"HowToStep","text":"Les placer sur la plaque du four, bien les espacer, les aplatir un peu."},{"@type":"HowToStep","text":"Faire cuire au four environ 10 minutes."},{"@type":"HowToStep","text":"Laissez les cookies refroidir."},{"@type":"HowToStep","text":"Et r\u00e9galez-vous!"}],"author":"Noelle_75","description":"sucre roux, chocolat, oeuf, farine, sucre, levure, beurre, sel","keywords":"Cookies de Pauline, cookie, sucre roux, chocolat, oeuf, farine, sucre, levure, beurre, sel,tr\u00e8s facile,bon march\u00e9,rapide","recipeCuisine":"","aggregateRating":{"@type":"AggregateRating","reviewCount":41,"ratingValue":4.5,"worstRating":0,"bestRating":5}}
  |  
  | </script>

```

Can something easily be done on Nextcloud side?

**Comment by mrzapp** [comment]
@sephrat This is a mistake on their side, and usually I'd recommend trying to to parse it on our end, but this is just a major screw-up, so it would be in their best interest to inform them.

This tag `<script type="application&#x2F;ld&#x2B;json">` is invalid.

They shoud use `<script type="application/ld+json">` instead.

Even search engines are not recognising their markup, it's completely ignored:

![image](https://user-images.githubusercontent.com/2747428/67664094-3fde6d00-f967-11e9-80e5-74072300c586.png)

**TL;DR**: Please inform the staff at marmiton.org that they need to correct their markup. Change `<script type="application&#x2F;ld&#x2B;json">` to `<script type="application/ld+json">`.

**Comment by sephrat** [comment]
Thanks for your help, I just reached out to them with this information.
Hope they will have a look into it.

**Comment by sephrat** [comment]
After ~1,5 months, still no feedback from them.  I just sent them another email to remind them of the issue.

Looking through Google documentation, I don't really get why the search results are messed up but their testing tool manages to retrieve relevant information: https://search.google.com/structured-data/testing-tool/u/0/?utm_campaign=devsite&utm_medium=jsonld&utm_source=recipe#url=https%3A%2F%2Fwww.marmiton.org%2Frecettes%2Frecette_cookies-de-pauline_64092.aspx 
It seems to me that Google manages to work around this encoding issue.

**Comment by mrzapp** [comment]
@sephrat we can try to decode HTML entities in our parser, but it might screw things up for other sites. I'll give it a shot in the next patch.

**Comment by mrzapp** [comment]
This fix will be pushed with 0.6.0
